xwiki-commons-core/xwiki-commons-extension/xwiki-commons-extension-api/src/main/java/org/xwiki/extension/repository/internal/local/LocalExtensionStorage.java

```diff
@@ -296,15 +300,61 @@ public void removeExtension(DefaultLocalExtension extension) throws IOException
         File descriptorFile = extension.getDescriptorFile();
 
         if (descriptorFile == null) {
-            throw new IOException("Exception does not exist");
+            throw new IOException(
+                String.format("Extension [%s] does not exist: descriptor file is null", extension.getId().getId()));
         }
 
-        descriptorFile.delete();
+        try {
+            Files.delete(descriptorFile.toPath());
+        } catch (FileNotFoundException e) {
+            LOGGER.warn(
+                "Couldn't delete the extension descriptor file [{}] when removing extension [{}], because it doesn't exist. Root error: [{}]",
+                descriptorFile.getAbsolutePath(), extension.getId().getId(), ExceptionUtils.getRootCauseMessage(e));
+        }
 
         DefaultLocalExtensionFile extensionFile = extension.getFile();
 
         if (extensionFile != null) {
-            extensionFile.getFile().delete();
+            try {
+                Files.delete(extensionFile.getFile().toPath());
+            } catch (FileNotFoundException e) {
+                LOGGER.warn("Extension file [{}] was not found while removing [{}] extension",
+                    extensionFile.getAbsolutePath(), extension.getId().getId());
+            }
+        }
+
+        // Get extension file descriptor path
+        Path extensionDescriptorFilePath = descriptorFile.toPath();
+
+        // Get the path to the folder that store the version of the extension being removed
+        Path extensionVersionFolderPath = extensionDescriptorFilePath.getParent();
+        try {
+            // Delete the extension version folder
```

Comment:
I would be more explicit in the comment on the fact that it will only do something if it's empty.